lib/puppet/provider/sysctl/augeas.rb

```diff
@@ -39,52 +39,73 @@ def self.sysctl_get(key)
 
   confine :feature => :augeas
 
-  def self.instances(reference_resource = nil)
+  def self.instances(reference_resources = nil)
     return @resource_cache if @resource_cache
 
-    resources = nil
+    resources = []
+    reference_resource_titles = reference_resources.map { |_ref_name, ref_obj| ref_obj.title }
+
+    sysctl_output = ''
+    if reference_resources
+      resource_dup = reference_resources.first.last.dup
+      files = ['/etc/sysctl.conf', resource_dup[:target]].compact
```

Comment:
Should it use `.uniq` as well in case the target is `/etc/sysctl.conf`?

Reply:
As a note, the consequence of this is that `puppet resource sysctl` will not return the same resources as when there's resources in the catalog pointing to other files than `/etc/sysctl.conf`.

Reply:
Yeah, fair enough on the `uniq`, I'll update that.

In terms of the `puppet resource sysctl`, that's pretty much how it always worked. Only the last resource item was popped off for assessment so I kept that behavior. I suppose that I could troll all of the resources since I have them.